src/main/resources/templates/c_plus_plus/test/Tests.py

```
@@ -4,23 +4,67 @@
+            if f.endswith(".cpp") or f.endswith(".c"):
+                name, _ = os.path.splitext(f)
+                rel_path = os.path.relpath(os.path.join(root, f), ".")
+                rel_path = rel_path.replace(os.sep, "/")
+                test_filenames[name] = rel_path
+
+    print(f"Found test: {test_filenames}")
+
+    # And adjust CMakeLists.txt accordingly
+    cmake_file = os.path.join('.', "CMakeLists.txt")
+    if not os.path.isfile(cmake_file):
+        raise FileNotFoundError(f"CMakeLists.txt not found")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)